### DISCUSSION TITLE
@Parcelize build error: This class does not have a constructor

### DISCUSSION THREAD

**Discussion by DonBin** [arkivanov/Decompose#478] [Q&A] [answered]
Hi guys,

I am using decompose for navigation, when I create a class

```kotlin
@Parcelize
    data class CustomConfig(
        val type: Type,
        val text: String = ""
    ) : Parcelable {
        enum class Type {
            LIST,
            DETAIL
        }
    }

```
as the parameter of navigation 
`private val navigation = StackNavigation<CustomConfig>()`
I got the error 
```

This class does not have a constructor
Class 'CustomConfig' is not abstract and does not implement abstract member public abstract fun describeContents(): Int defined in android.os.Parcelable
```

and also I import 

```
import com.arkivanov.essenty.parcelable.Parcelable
import com.arkivanov.essenty.parcelable.Parcelize
```

Is there someone who can help ? 
thanks

**Comment by arkivanov** [comment]
Have you applied `kotlin-parcelize` Gradle plugin in that Gradle module?

**Reply by DonBin** [reply]
Yes, I think I have tried, all this happens in the shared `commonMain`
In the `gradle/libs.versions.toml` file

`kotlin-parcelize = { id = "org.jetbrains.kotlin.plugin.parcelize", version.ref = "kotlin-parcelize" }`

and in shared build.gradle

```kotlin
plugins {
    alias(libs.plugins.kotlin.multiplatform)
    alias(libs.plugins.android.library)
    alias(libs.plugins.kotlin.parcelize)
}
```
sync, then I got another error

```
Error resolving plugin [id: 'org.jetbrains.kotlin.plugin.parcelize', version: '1.9.1']
> The request for this plugin could not be satisfied because the plugin is already on the classpath with an unknown version, so compatibility cannot be checked.
```

Is there something I did wrong ?

**Reply by arkivanov** [reply]
Ok, perhaps that `alias` syntax comes from [decompose-multiplatform-template](https://github.com/arkivanov/decompose-multiplatform-template)? I never used `alias`, Gradle plugins usually applied using `id(<plugin-name>)`. I filed an [issue](https://github.com/arkivanov/decompose-multiplatform-template/issues/5) to update the template.

Please try applying plugins using `id`. You can refer to the following doc: https://docs.gradle.org/current/userguide/plugins.html

**Reply by arkivanov** [reply, marked as answer]
I think you can just write `id(kotlin-parcelize)`. I would recommend using `id` for all plugins.

**Reply by DonBin** [reply]
It works by using 
```kotlin
plugins {
    alias(libs.plugins.kotlin.multiplatform)
    alias(libs.plugins.android.library)
    id("kotlin-parcelize")
}
```
thanks [arkivanov](https://github.com/arkivanov), I am a newer on Android, not that familiar with gradle configuration, really appreciate your help.

**Reply by arkivanov** [reply]
You are welcome!
